git_greppers/utils.sh

```
@@ -0,0 +1,127 @@
+  local repo_identifier=""
+  if [[ $repo_type == "github" ]]; then
+    repo_identifier=$(echo $url | sed -e 's/https:\/\/github.com\///')
+    # Send debug output to stderr
+    debug 3 "GitHub repo identifier: $repo_identifier" >&2
+  elif [[ $repo_type == "gitlab" ]]; then
+    # Handle both https://gitlab.com/org/repo and git@gitlab.com:org/repo formats
+    repo_identifier=$(echo $url | sed -e 's/https:\/\/gitlab.com\///' -e 's/git@gitlab.com://')
+    # Send debug output to stderr
+    debug 3 "GitLab repo identifier: $repo_identifier" >&2
+    # Store the repo identifier in a global variable so it can be accessed by the fetch_gitlab_mr_comments function
+    GITLAB_REPO_IDENTIFIER="$repo_identifier"
```

Shellcheck (0.10.0)

[warning] GITLAB_REPO_IDENTIFIER appears unused. Verify use (or export if used externally).

(SC2034)